build_dependency_tree.py

```diff
@@ -3,76 +3,202 @@
+                    # (required by standards)
+                    re_match = include_statement_pattern.match(line.split()[1])
                     # We only expect one entry in the list
-                    if not re_match[0]: # first matching group is empty
+                    if not re_match:  # first matching group is empty
+                        print(f'No header files found in line: {line}')
                         continue
+                    elif not re_match.group(1):
+                        # Check if match with `<...>` is None
+                        print(f"""Header file found
+                                {str(file.absolute())}:{re_match.group(2)}""")
+                        include_statements.append(
+                                extract_header_file_name(
+                                    re_match.group(2)))
                     else:
-                        include_statements.append(re_match[0])
-                else:
-                    continue
+                        # since we have a match and first group is non-empty we
+                        # apend
+                        print(f"""Header file found
+                                {str(file.absolute())}:{re_match.group(1)}""")
+                        include_statements.append(
+                                extract_header_file_name(
+                                    re_match.group(1)))
         return include_statements
+
+
+def output_dependency_tree_to_dot_file(
+        dep_tree: Dict[List[Union[Path, str]], str],
+        output_name: Optional[Union[Path, str]]):
+    """
+    Goes through dictionary of files gathered from recursive search of
+    directory and prints all the nodes it will be attached to in file
+    named output_name.
+    Output is formated to be compatible with the DOT command line program:
+    https://graphviz.org/doc/info/lang.html
+
+    :param dep_tree:    dictionary with files in project as keys and list
+                        of their dependencies as value
+    :param output_name: name of output file or path to output file
+    """
+    if not isinstance(dep_tree, dict):
+        # For cleaner debugging messages, only keep while still developing
+        # program
+        print("None dictionary object passed. Exiting...")
+        quit()
+
+    if not output_name:
+        output_name = Path('./includes_tree_output.dot').absolute()
+
+    print(f'Output file is: {str(output_name)}')
+
+    # TODO: Need to colorize output.
+    #       E.g. header files are blue, translation units red
+    #       Standard Template Library files are different shaped bubbles, etc
+    #       Of course this will require a key of some sort to be documentd
+    #       somewhere
+    with open(str(output_name.absolute()), 'w') as f:
+        f.write("graph {\n")
+        for key, dep_list in dep_tree.items():
+            for entry in dep_list:
+                f.write(f"\"{key}\" -- \"{entry}\";\n")
```

Comment:
```suggestion
       _ = [[f.wtite(f'"{key}" -- "{entry}";\n') for entry in dep_list] for key, dep_list in dep_tree.items()]
```
We love that comprehension. But since we aren't using this anywhere you can leave it in the for loops. you can switch to `f'...'` tho to make using double quotes more intuitive